exporter/ethstore.go

```diff
@@ -51,35 +51,65 @@ func StartEthStoreExporter(bnAddress string, enAddress string, updateInterval, e
 	// Reexport days if specified
 	if startDayReexport != -1 && endDayReexport != -1 {
 		for day := startDayReexport; day <= endDayReexport; day++ {
-			err := ese.exportDay(strconv.FormatInt(day, 10), true)
+			err := ese.reexportDay(strconv.FormatInt(day, 10))
 			if err != nil {
-				logger.WithError(err).Errorf("error reexporting eth.store day %d in database", day)
+				utils.LogError(err, fmt.Sprintf("error reexporting eth.store day %d in database", day), 0)
 				return
 			}
 		}
 	}
 
 	ese.Run()
-
 }
 
-func (ese *EthStoreExporter) exportDay(day string, shouldClearDay bool) error {
-	ethStoreDay, validators, err := ese.getStoreDay(day)
+func (ese *EthStoreExporter) reexportDay(day string) error {
+	tx, err := ese.DB.Beginx()
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	ese.prepareClearDayTx(tx, day)
 	if err != nil {
 		return err
 	}
 
+	ese.prepareExportDayTx(tx, day)
+	if err != nil {
+		return err
+	}
+
+	return tx.Commit()
+}
+
+func (ese *EthStoreExporter) exportDay(day string) error {
 	tx, err := ese.DB.Beginx()
 	if err != nil {
 		return err
 	}
 	defer tx.Rollback()
 
-	if shouldClearDay {
-		_, err = tx.Exec(`DELETE FROM eth_store_stats WHERE day = $1`, ethStoreDay.Day)
-		if err != nil {
-			return err
-		}
+	ese.prepareExportDayTx(tx, day)
+	if err != nil {
+		return err
+	}
+
+	return tx.Commit()
+}
+
+func (ese *EthStoreExporter) prepareClearDayTx(tx *sqlx.Tx, day string) error {
+	dayInt, err := strconv.ParseInt(day, 10, 64)
+	if err != nil {
+		return err
+	}
+	_, err = tx.Exec(`DELETE FROM eth_store_stats WHERE day = $1`, dayInt)
+	return err
+}
+
+func (ese *EthStoreExporter) prepareExportDayTx(tx *sqlx.Tx, day string) error {
```

Comment:
This does not only prepare the Tx, it also commits it:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/11e87b40caa41797c93b8e44bfd48315f4deb171/exporter/ethstore.go#L206-L210

Please adapt `prepareExportDayTx` so that it acts like `prepareClearDayTx`.

Reply:
Good find, commit was deleted.